app/controllers/catalog_controller.rb

```diff
@@ -933,6 +936,26 @@ def retrieve_and_set_item_id
   end
 
   #
+  # Ensures that the item is indexed in Solr and displays an appropriate response otherwise
+  #
+  def check_item_indexed
+    item = nil
+    item = Item.find(params[:id]) unless params[:id].blank?
```

Comment:
I think that you are doing a lot of unnecessary checks here. I don't think you need to assign the item `nil` or check that `:id` is blank. The `find` method should raise `ActiveRecord::RecordNotFound` if there is no record found (including handing the find method an empty string or `nil`), so you should be able to wrap it in a `begin ... rescue` block with just `item = Item.find(params[:id])`, which would also mean you wouldn't need `unless item.nil?`